python/graphstorm/run/launch.py

```diff
@@ -733,8 +913,56 @@ def signal_handler(sig, frame): # pylint: disable=unused-argument
         logging.error("Remapping task failed")
         sys.exit(-1)
 
+def get_ip_config(ip_config, workspace):
+    """ Get IP configurations
+
+    Parameters
+    ----------
+    ip_config : str
+        The path to the IP config file.
+    workspace : str
+        The path of the workspace
+
+    Returns
+    -------
+    str : the absolute path of IP configuration file.
+    list of tuple : the IP address and ports of the hosts.
+    """
+    # Get the IP addresses of the cluster.
+    # The path to the ip config file can be a absolute path or
+    # a relative path to the workspace
+    hosts = []
+    if ip_config is not None:
+        ip_config = ip_config if os.path.isabs(ip_config) else \
+                os.path.join(workspace, ip_config)
+        assert os.path.isfile(ip_config), \
+                f"IP config file must be provided but got {ip_config}"
+        with open(ip_config, encoding='utf-8') as f:
+            for line in f:
+                result = line.strip().split()
+                if len(result) == 2:
+                    ip = result[0]
+                    port = int(result[1])
+                    hosts.append((ip, port))
+                elif len(result) == 1:
+                    ip = result[0]
+                    port = get_available_port(ip)
+                    hosts.append((ip, port))
+                else:
+                    raise RuntimeError("Format error of ip_config.")
+    else:
+        # The user doesn't provide an IP config file. This means we are going to
+        # run the training job in the local machine. We should create a temporary
+        # IP config file.
+        with tempfile.NamedTemporaryFile(delete=False) as fp:
```

Comment:
Who will remove this file?

Reply:
And why do we still need this temp file as we already know the ip address?

Reply:
DistDGL needs this file.

Reply:
right now it's not deleted. this is a temporary file. should we delete it?